docs/topics/multiplatform/multiplatform-compatibility-guide.md

```diff
@@ -490,3 +490,200 @@ projects.
+                   implementation(projects.myLibrary.ktor)
+               }
+           }
+   
+           // 'jvmOkioTest' source set contains okio-specific integration tests
+           val jvmOkioTest by getting {
+               dependencies {
+                   implementation(libs.okio)
+                   implementation(projects.myLibrary.okio)
+               }
+           }
+       }
+   }
+    ```
+
+   </tab>
+   <tab title="Now" group-key="now">
+   
+    ```kotlin
+   // Project ':myLibrary:integrationTests'
+   kotlin {
+      jvm {
+         val ktorIntegrationTest by compilations.creating
+         val okioIntegrationTest by compilations.creating
+   
+         // Associate with test compilation to see tests from commonTest
+         ktorIntegrationTest.associateWith(compilations.getByName("test"))
+         okioIntegrationTest.associateWith(compilations.getByName("test"))
+   
+         // Create a Test Run to for each integration test
+         testRuns {
+            create("ktorIntegration") { setExecutionSourceFrom(ktorIntegrationTest) }
+            create("okioIntegration") { setExecutionSourceFrom(okioIntegrationTest) }
+         }
+      }
+   
+      sourceSets {
+         val commonTest by getting {
+            dependencies {
+               implementation(kotlin("test"))
+               implementation(projects.myLibrary)
+            }
+         }
+   
+         val jvmKtorIntegrationTest by getting {
+            dependencies {
+               implementation(kotlin("test"))
+               implementation(libs.ktor)
+               implementation(projects.myLibrary.ktor)
+            }
+         }
+   
+         val jvmOkioIntegrationTest by getting {
+            dependencies {
+               implementation(kotlin("test"))
+               implementation(libs.okio)
+               implementation(projects.myLibrary.okio)
+            }
+         }
+      }
+   }
+    ```
+
+   </tab>
+   </tabs>
+
+
+
+2. Extracting new target into a separate Gradle project
+
+   <tabs group="build-script">
+   <tab title="Before" group-key="before">
+
+   ```kotlin
+   // Project ':myUILibrary'
+   kotlin {
+       jvm("jvmCompose") // !!!
+       jvm("jvmSwing")   // !!!
+   
+       sourceSets {
+           val commonMain by getting
+           val commonJvmMain by sourceSets.creating {
+               dependsOn(commonMain)
+               dependencies {
+                   // shared dependencies
+               }
+           }
+           val jvmComposeMain by getting {
+               dependsOn(commonJvmMain)
+               dependencies {
+                   // compose specific dependencies
+               }
+           }
+           val jvmSwingMain by getting {
+               dependsOn(commonJvmMain)
+               dependencies {
+                   // swing specific dependencies
+               }
+           }
+       }
+   }
+   ```
+
+   </tab>
+   <tab title="Now" group-key="now">
+
+```kotlin
```

Comment:
I'm a bit conernced that this example is neither concise, nor complete. I think we have to make a firm decision: either we're providing complete examples, or we don't provide them at all and just assume that the users that had multiple same targets are advanced enough to handle this refactoring on their own.

I'm leaning towards the latter. Mostly because it's so hard to give a proper minimal migration example -- as it involves multiple files in the project, the only solution I see is an actual GitHub project with "migration commit", but IMO it's too much for this case (but we can expand it, if we see demand/questions)

Reply:
Agree with @dsavvinov here. Also, it would be nice to include a paragraph about when to choose separate compilations over separate Gradle projects. Adding a separate Gradle project is usually more involved. Unless there's something fundamental it unlocks, I would even not mention it.